src/core/global.c

```diff
@@ -190,6 +179,10 @@ static void nn_global_handler (struct nn_fsm *self,
 static void nn_global_shutdown (struct nn_fsm *self,
     int src, int type, void *srcptr);
 
+/*  Socket holds. */
+static int nn_global_hold_socket(struct nn_sock **sockp, int s);
+static int nn_global_hold_socket_locked(struct nn_sock **sockp, int s);
+static void nn_global_rele_socket(struct nn_sock *);
```

Comment:
Actually, the "hold" terminology I picked up from kernel programming, where it is incredibly common.  I don't know how else to describe what I'm doing.